solutions/instances/main.tf

```diff
@@ -151,7 +152,25 @@ module "kms" {
 # COS
 #######################################################################################################################
 
-module "cos" {
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_authorization_policy" {
+  depends_on      = [ibm_iam_authorization_policy.policy]
+  count           = var.skip_cos_kms_auth_policy ? 0 : 1
+  create_duration = "30s"
+}
+
+# Create IAM Authorization Policy to allow COS to access KMS for the encryption key
+resource "ibm_iam_authorization_policy" "policy" {
+  count                       = var.skip_cos_kms_auth_policy ? 0 : 1
+  source_service_name         = "cloud-object-storage"
+  source_resource_instance_id = local.cos_instance_guid
+  target_service_name         = "hs-crypto" # Currently hard-coded need to update
```

Comment:
how do you plan on getting this service name? Do we need to ask for the CRN instead and parse the service and guid from it?

Reply:
Yes, I plan to ask for CRN and then parse the service and guid from it. Then I will need to do perform changes from `kms_guid` -> `kms_crn`.